### ISSUE TITLE
When zoomed in panning up and down triggers onSwipeToClose

### ISSUE THREAD

**Issue by EldinHb** [Flair-Dev/react-native-awesome-gallery#91]
When zoomed in panning up and down triggers onSwipeToClose too quickly, because of this it is impossible to pan up and down. And when returning early in onSwipeToClose I can keep panning to what seems infinity in any direction.

Anyone else experiencing the same problem? 

This is the code:

```typescript
<View>
    <Gallery
         data={[img]}
         onSwipeToClose={() => router.back()}
         keyExtractor={x => x.thumb}
         renderItem={(item) => (
	         <Image
		         style={{
			objectFit: 'contain',
			flex: 1
		}}
		source={{
			uri: item.item.fullsize
		}}
	/>
     )}
/>
</View>
```

iOS 18.2
Expo 52
RN Gesture handler 2.20.2
Reanimate 3.16.1

**Comment by kolking** [comment]
Yes, this is quite annoying behaviour. I believe the `onSwipeToClose` should never be triggered if scale > 1. Currently it's not triggered only if the scaled image's height exceeds the screen height. 

- `scale = 1`: can be swiped down to close
- `scale = 2`: panning down will close gallery (this is incorrect)
- `scale = 3`: won't close because now it exceeds the screen height

scale = 1 | scale = 2 | scale = 3
---|---|---
![Image](https://github.com/user-attachments/assets/568d61a7-e89e-4be2-9f14-14c8f0a835ee) | ![Image](https://github.com/user-attachments/assets/a166e29e-077c-4324-9f56-226d4d484515) | ![Image](https://github.com/user-attachments/assets/12e9b228-cad1-443c-976b-9b5cf1b1866e)

**Comment by oeddyo** [comment]
this seems to be fixed. What version are you using? @kolking

**Comment by kolking** [comment]
@oeddyo the latest `v0.4.2`, I cannot confirm it's fixed.
